repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasRelationshipStoreV2.java

```diff
@@ -376,16 +373,17 @@ public void deleteById(String guid, boolean forceDelete) throws AtlasBaseExcepti
                 );
             }
         }
-
-        sendNotifications(entityRetriever.mapEdgeToAtlasRelationship(edge), OperationType.RELATIONSHIP_DELETE);
+        addAtlasObjectIdToVertexIdMappingForEndVertices(edge);
+        AtlasRelationship deletedRelationship = entityRetriever.mapEdgeToAtlasRelationship(edge);
+        sendNotifications(deletedRelationship, OperationType.RELATIONSHIP_DELETE);
+        RequestContext.get().addToDeletedRelationships(deletedRelationship);
```

Comment:
I think notifying should be the final step.? Will that make any difference, if we update the context first and then notify ?